gpflow/posterior.py

```diff
@@ -0,0 +1,194 @@
+# Copyright 2016-2020 The GPflow Contributors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Tuple
+
+import numpy as np
+import tensorflow as tf
+
+from . import covariances, kernels
+from .base import Module, Parameter
+from .conditionals.util import expand_independent_outputs, mix_latent_gp
+from .config import default_float, default_jitter
+from .models.model import MeanAndVariance
+
+
+class DiagNormal(Module):
+    def __init__(self, q_mu, q_sqrt):
+        self.q_mu = q_mu  # [M, L]
+        self.q_sqrt = q_sqrt  # [M, L]
+
+
+class MvnNormal(Module):
+    def __init__(self, q_mu, q_sqrt):
+        self.q_mu = q_mu  # [M, L]
+        self.q_sqrt = q_sqrt  # [L, M, M], lower-triangular
+
+
+class Posterior(Module):
+    def __init__(self, kernel, iv, q_mu, q_sqrt, whiten=True, mean_function=None):
```

Comment:
Nit: everywhere else in the codebase we use `inducing_variable` and not `iv`, it would be nice to be consistent.